### PR TITLE
feature/59771 surface input assistant data

### PR DESCRIPTION
**Pull request by GuyHarwood**
- surface input assistant data in core database
- migrate retro input assistant to new table 
- update associated services and backend workers
- refactor PS report
- bridge gap of retro input assistant inclusion in PS report

### CODE REVIEW

File: pupil-spa/src/app/services/pupil-prefs/pupil-prefs.service.spec.ts [hunk 1]
```diff
@@ -9,6 +9,8 @@ import { AuditService } from '../audit/audit.service';
 import { QuestionService } from '../question/question.service';
 import { QuestionServiceMock } from '../question/question.service.mock';
 import { AccessArrangements } from '../../access-arrangements';
+import { first } from 'rxjs/operators'
```
Comment (jon-shipley) [inline pupil-spa/src/app/services/pupil-prefs/pupil-prefs.service.spec.ts:12]: I don't see this operator being used in the diff?

**Review by jon-shipley — APPROVED** [review]
LGTM  👍

File: pupil-spa/src/app/services/pupil-prefs/pupil-prefs.service.spec.ts (hunk 1 above)
Comment (GuyHarwood) [reply pupil-spa/src/app/services/pupil-prefs/pupil-prefs.service.spec.ts:12]: i will look into this and provide an update...

Comment (GuyHarwood) [reply pupil-spa/src/app/services/pupil-prefs/pupil-prefs.service.spec.ts:12]: removed.  Ng linter is not picking this up

**Pull request merged**